drivers/eeprom/Kconfig

```diff
@@ -68,7 +68,8 @@ config EEPROM_AT24
 	bool "Atmel AT24 (and compatible) I2C EEPROM support"
 	default y
 	depends on DT_HAS_ATMEL_AT24_ENABLED
-	select I2C
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ATMEL_AT24),i2c)
+	select I3C if $(dt_compat_on_bus,$(DT_COMPAT_ATMEL_AT24),i3c)
```

Comment:
This driver does not use I3C, remove the select I3C

Reply:
Hi @XenuIsWatching @henrikbrixandersen, could you pls revisit the comment above

Reply:
This is still wrong, and the second paragraph is off the rails with how i3c/i2c works... remove it

You even have the eeprom on an i2c bus above with the second reg set to 0x00. This makes it seen as if it's on a i2c bus.

Reply:
Thanks for your feedback, so what would use suggest to build this driver for the boards that have eeprom connected to i3c bus (because the `CONFIG_I3C` is not set --> I3C shim driver is not built):
1. No use it at all
2. Or application should set `CONFIG_I3C=y` directly

Reply:
These AT24 EEPROMs are I2C EEPROMs, not I3C EEPROMs and the driver uses (and should continue to use) I2C controller API calls, not I3C controller API calls.

Since the Zephyr I3C controller API struct includes the Zephyr I2C controller API struct, I really see no good way of automatically selecting I3C besides how it is done here.

Reply:
@dcpleung Any opinion on this?

Reply:
> Thanks for your feedback, so what would use suggest to build this driver for the boards that have eeprom connected to i3c bus (because the `CONFIG_I3C` is not set --> I3C shim driver is not built):
> 
> 1. No use it at all
> 2. Or application should set `CONFIG_I3C=y` directly

I am even using this same EEPROM driver in my application with an I3C bus... it works as is
You should be having `CONFIG_I3C=y` set directly in the project `prj.conf` 😖 This is how you should have already had it.

Reply:
Well, of course it works if you set it explicitly. The consensus is for drivers to select the bus which they are on.

In theory, all drivers currently selecting I2C should conditionally select I3C as well, if their device is on an I3C bus.

Reply:
yes... but it is defined as on an `I2C` bus. There are dozens and dozens of I2C drivers out there. I do not find it ideal to go through and change EVERY SINGLE ONE to have it include `I3C`.... but i do understand your point about how the bus driver is selected though an `if I3C`. It may be worth investigating if that should be a `if I3C || I2C` instead...
https://github.com/zephyrproject-rtos/zephyr/blob/a7dccc4475f342c9b89f484f796001bc1c6d8a97/drivers/i3c/Kconfig#L12